src/components/FilmList/FilmList.tsx

```diff
@@ -1,28 +1,71 @@
+import { useEffect, useState } from "react";
+import { filterByGenre, sorting } from "../../helpers";
 import { Film } from "../../interfaces";
 import { FilmCard } from "../FilmCard";
 
 import "./style.scss";
 
 interface FilmListProps {
+  selectedGenre: string;
+  sortBy: string;
+  onMovieItemClick(film: Film): void;
   movies: Film[];
-  openModal(type: string, filmID: string): void;
+  openModal(type: string, film: Film): void;
 }
 
-const FilmList: React.FC<FilmListProps> = ({ movies, openModal }) => {
-  const filmsCards: any = movies.map((el) => {
-    return <FilmCard film={el} key={el.id} openModal={openModal} />;
+const FilmList: React.FC<FilmListProps> = ({
+  movies,
+  openModal,
+  selectedGenre,
+  sortBy,
+  onMovieItemClick,
+}) => {
+  const [moviesConfigs, setMoviesConfigs] = useState({
+    movies: sorting(filterByGenre(movies, selectedGenre), sortBy),
+    sort: selectedGenre,
+    genre: sortBy,
   });
 
-  const noFilm = <div>No movie found</div>;
+  useEffect(() => {
+    setMoviesConfigs((prevState) => {
+      if (prevState.genre !== selectedGenre) {
+        return {
+          ...prevState,
+          genre: selectedGenre,
+          movies: sorting(filterByGenre(movies, selectedGenre), sortBy),
+        };
+      }
+
+      return {
+        ...prevState,
+        genre: selectedGenre,
+        movies: sorting(prevState.movies, sortBy),
+      };
+    });
+  }, [movies, selectedGenre, sortBy]);
+
```

Comment:
you only covered one case.
But there are few others, like the previous genre was 'ALL'..
and the case you covered, could be done with just two useMemo and move this out of FilmList component
```
const filteredMoviesByGenre = useMemo(()=> filterByGenre(movies, selectedGenre), [movies, selectedGenre]);

const sortedMovies = useMemo(()=> sorting(filteredMoviesByGenre , sortBy), [filteredMoviesByGenre, sortBy]))
```